src/ai/ai.cpp

```diff
@@ -678,65 +678,86 @@ void update_ai_econ_construction(sys::state& state) {
-					auto end = start + state.world.national_identity_get_political_party_count(identity);
-
-					for(int32_t i = start; i < end && !target; i++) {
-						auto pid = dcon::political_party_id(uint16_t(i));
-						if(politics::political_party_is_active(state, pid) && (state.culture_definitions.governments[gov].ideologies_allowed & ::culture::to_bits(state.world.political_party_get_ideology(pid))) != 0) {
-
-							for(auto pi : state.culture_definitions.party_issues) {
-								auto issue_rules = state.world.political_party_get_party_issues(pid, pi).get_rules();
-								if((issue_rules & issue_rule::build_factory) != 0) {
-									target = pid;
-									break;
-								}
+		bool can_appoint = [&]() {
+			if(!politics::can_appoint_ruling_party(state, n))
+				return false;
+			auto last_change = state.world.nation_get_ruling_party_last_appointed(n);
+			if(last_change && state.current_date < last_change + 365)
+				return false;
+			if(politics::is_election_ongoing(state, n))
+				return false;
+			return true;
+			/*auto gov = state.world.nation_get_government_type(source);
+			auto new_ideology = state.world.political_party_get_ideology(p);
+			if((state.culture_definitions.governments[gov].ideologies_allowed & ::culture::to_bits(new_ideology)) == 0) {
+				return false;
+			}*/
+		}();
+
+		// Before, the AI would keep appointing parties repeatedly
+		// causing massive incursions of militancy amongst the population
+		// so stop appointing political parties if we have too much militancy
+		// so we stop at 75% of the military cap for nationalist movements, to not trigger them
+		// hopefully...
+		auto militancy = state.world.nation_get_demographics(n, demographics::militancy);
+		auto total_pop = state.world.nation_get_demographics(n, demographics::total);
+		if(total_pop / militancy >= state.defines.nationalist_movement_mil_cap * 0.75f) {
+			can_appoint = false;
+		}
+
+		if(can_appoint) {
+			auto gov = n.get_government_type();
+			auto identity = n.get_identity_from_identity_holder();
+			auto start = state.world.national_identity_get_political_party_first(identity).id.index();
+			auto end = start + state.world.national_identity_get_political_party_count(identity);
+			dcon::political_party_id target;
+			if(current_iscore < 5 && (rules & issue_rule::build_factory) == 0) { // try to jumpstart econ
+				for(int32_t i = start; i < end && !target; i++) {
+					auto pid = dcon::political_party_id(uint16_t(i));
+					if(politics::political_party_is_active(state, pid) && (state.culture_definitions.governments[gov].ideologies_allowed & ::culture::to_bits(state.world.political_party_get_ideology(pid))) != 0) {
+						for(auto pi : state.culture_definitions.party_issues) {
+							auto issue_rules = state.world.political_party_get_party_issues(pid, pi).get_rules();
+							if((issue_rules & issue_rule::build_factory) != 0) {
+								target = pid;
+								break;
 							}
 						}
 					}
+				}
+			} else {
```

Comment:
Why do this? Appointing any party will make some people mad, right. And there seems to be no limit to how often you are doing this other than possibility, so even if it makes a minority of people angry, you will be making some people angrier and angrier for no reason. I suggest using nation_get_ruling_party_last_appointed to make this condition happen less frequently. Once every 5 or 10 years, perhaps

Reply:
it is worth the cost for the return on investment, the ai sacrifices short term militancy for long term militancy decrease

Reply:
fine changing this

Reply:
I said that you *could* do it, just less often than once a year. You didn't have to take it out completely